electrum_dash/transaction.py

```diff
@@ -647,11 +647,11 @@ def read_vds(cls, vds, alone_data=False, tx=None) -> None:
             tx._version = header
 
         n_vin = vds.read_compact_size()
-        if n_vin < 1:
+        if n_vin < 1 and tx._tx_type not in (9,):
```

Comment:
Why do we need this check?

Reply:
As I understand it, there may be no inputs for AssetUnlock. That's what the check is for.